src/Servers/Kestrel/Transport.Sockets/src/SocketTransportOptions.cs

```diff
@@ -16,6 +16,14 @@ public class SocketTransportOptions
         /// </remarks>
         public int IOQueueCount { get; set; } = Math.Min(Environment.ProcessorCount, 16);
 
+        /// <summary>
+        /// Wait until there is data available to allocate a buffer. Set this to true to reduce memory used for idle connections.
```

Comment:
Setting this to false can also increase throughput at the cost of increased memory usage.